### DISCUSSION TITLE
What are the client and the server bundles? How do they work?

### DISCUSSION THREAD

**Discussion by sumdook2** [vercel/next.js#14525] [Help] [answered]
What are the different client and the server bundles? How do they work? How do they affect performance?

**Comment by jamesmosier** [comment, marked as answer]
Client bundles are served to your users via their browser. Performance matters for the client bundle which Next does a great job optimizing. 

Server bundles are never "seen" by the user nor are they ever shipped to the end user in their browser. They are run in NodeJS on your server/hosting environment.

**Reply by sumdook2** [reply]
Thank you so much. Your are a lifesaver! I have been losing my mind over this for weeks!!

**Reply by prichodko** [reply]
@jamesmosier - thank you for you answer.

Can you please elaborate a bit more on the server bundles? What does it mean exactly that they are run in Node.js? What could happen if they are too large?

**Reply by jamesmosier** [reply]
Next is a server rendered React framework. Therefore part of your application runs on the server, which uses Node.js. 

So when you execute `next start` that is starting a Node.js server. Then when a users requests your page it hits your Node server, executes some JavaScript on the server (this would be your server code, not really bundles, which the user never sees and file size does not matter because it is never served to a users browser) and then it returns the result (i.e. your webpage) to your user along with clientside JavaScript bundles.
